### DISCUSSION TITLE
Streaming audio from various clients to the snapcast server

### DISCUSSION THREAD

**Discussion by alkenerly** [snapcast/snapcast#1195] [Q&A]
I came across this project a week ago and have since used it to create a "DIY Sonos" multi-room solution in my home. I absolutely love it and am very impressed with the quality and extensibility of this project. Big thanks to badaix and all the other contributors. 

Now that it is setup, it leaves me wondering about what all I can do with it. Particularly, I wonder if anyone in this community is aware of methods to send audio data to the snapserver's TCP server source from things like a Windows PC or Android device. Like an android app that could listen to the microphone, packet up the data into the proper format and send it to the snapserver like its a PA system or something. Similarly, configuring an Windows audio device that could just send everything going through it to the snapserver over TCP. It would make it user friendly to source audio from youtube, netflix, etc. 

Relevant other discussions:
https://github.com/badaix/snapcast/discussions/875 (May 2021)

**Comment by StuartIanNaylor** [comment]
I am a noob also, but my 1st thought why not a snapserver on other OS with just a loopback, you just have a client with the other OS hostname playing into local snapserver ... ?
Not sure what OS snapserver builds on though, but the answer could be snapcast.

I am planning to do this with a stereo pair with a BT reciever and have a local snapcast server and x2 clients on each speaker with one assisgned to local hostname and the other to a central server.
I have to test it as hoping the sync will be tight enough not to give a 'flanger' like effect.

But other solutions
https://github.com/stijnvdb88/Snap.Net/blob/master/Snap.Net.Broadcast/README.md
https://github.com/duncanthrax/scream `Virtual network sound card for Microsoft Windows`
https://github.com/duncanthrax/scream/tree/master/Receivers/unix
Is one for windows that lets you just select a soundcard.
https://vac.muzychenko.net/en/download.htm loopback for windows

Mac 'I know nothing'
https://github.com/SimulPiscator/vpcm ? But has the links to https://github.com/mattingalls/Soundflower and https://github.com/ExistentialAudio/BlackHole is a Mac Loopback.

Would be good if others post who maybe not as noob as I, as do any have native stream to or play to tcp file socket?
